src/xcode/ENA/ENA/Source/Scenes/Home/HomeInteractor.swift

```diff
@@ -96,16 +93,16 @@ final class HomeInteractor: RequiresAppDependencies {
 	}
 
 	private func observeRisk() {
-		riskConsumer.didChangeLoadingStatus = { isLoading in
-			self.updateAndReloadRiskLoading(isRequestRiskRunning: isLoading)
+		riskConsumer.didChangeActivityState = { state in
+			self.updateAndReloadRiskCellState(to: state)
```

Comment:
I think there is a retain cycle here and a potential memory leak. HomeInteractor -> riskConsumer -> didChangeActivityState -> HomeInteractor

Reply:
Good point, I made the reference to self (home interactor) weak here so the retain cycle should not exist any longer! 👍